### PR TITLE
[issue#42_MnistEvaluator] Work in progress

### LINKED ISSUE #42: Написать класс-оценщик для обученной модели и получить метрики
Чтобы понять с каким качеством работает твоя обученная модель, нужно написать класс, который внутри себя будет использовать класс Inference, но хранить значения: accuracy, precision, recall и f1-macro. 
Нужно, чтобы класс хранил информацию как для каждой картинки в отдельности, так и умел считать средние показатели метрик. Удобнее всего хранить результаты в csv-формате. 

В результате выполнения задачи должны быть: 
- класс оценщик 
- ноутбук, который оценивает качество модели на тестовой выборке и сохраняет csv-файлы по указанному пути.

### PR DESCRIPTION
**Pull request by MarinaKupetskova**

### CODE REVIEW

File: object_detection/mnist_evaluation.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import numpy as np
+import pandas as pd
+from sklearn.metrics import classification_report
+from tqdm import tqdm
+
+
+class MnistEvaluator:
+    def __init__(self, inference: object, data: list):
```
Comment (serykhelena) [inline object_detection/mnist_evaluation.py]: сюдв вместо object можно прям тип mnist_inference.Inference написать (только надо ещё импорт)

File: object_detection/mnist_evaluation.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        self._infer = inference
+        self._data = data
+        self._predictions = None
```
Comment (serykhelena) [inline object_detection/mnist_evaluation.py]: логгера не хватает

Comment (serykhelena) [inline object_detection/mnist_evaluation.py]: можно вместо None, сразу создать пустой датафрейм с нужными колонками. Например, 
```
self._predictions = pd.Dataframe([], columns=["true_label", "pred_label"])
```

File: object_detection/mnist_evaluation.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    def evaluate(self):
+        pred = []
+        labels = []
+        for img, label in tqdm(self._data):
+            img = np.array(img)
+            pred.append(self._infer.get_prediction(img))
+            labels.append(label)
+
+        df_predicion = pd.DataFrame(labels, columns=["true_label"])
```
Comment (serykhelena) [inline object_detection/mnist_evaluation.py]: и сюда уже сразу в `self._predictions` добавлять данные, при этом можно сделать из 2-х листов матрицу с 2-мя колонками через `np.column_stack((labels, pred))`

File: object_detection/mnist_evaluation.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        df_predicion["pred_label"] = pred
+        self._predictions = df_predicion
+        return df_predicion
+
+    def classification_report(self):
+        if self._predictions is None:
```
Comment (serykhelena) [inline object_detection/mnist_evaluation.py]: и здесь вместо проверки на None, можно делать 
```
if self._prediction.empty: 
    ...
```

File: .gitignore [hunk 5]
```diff
@@ -140,3 +140,6 @@ checkpoints
 
 # Tensorboard runs
 runs
+
+# mnist predictions content
+*.cvs
```
Comment (serykhelena) [inline .gitignore]: давай сделаем папки results/mnist/ и в неё уже будем сохранять результаты оценщика. А саму папку results целиком игнорим

**Review by serykhelena — CHANGES_REQUESTED** [review]
Давай ещё при сохранении csv файлов в название добавлять дату, когда этот файл сохранялся чекрез модуль datetime
Формат даты YYYY-MM-DD 
И ещё classification_report тоже надо сохранять в csv файл

**Review by serykhelena — APPROVED** [review]
